backend/app/Main.hs

```diff
@@ -24,35 +25,38 @@ redisConfig = defaultConnectInfo
 logType :: LogType
 logType = LogStdout defaultBufSize
 
+corsMiddleware :: Application -> Application
+corsMiddleware = cors $ const $ Just policy
+  where
+    policy =
+      CorsResourcePolicy
+        { corsOrigins = Just (["http://127.0.0.1:3000"], True),
+          corsMethods = ["POST"],
+          corsRequestHeaders = ["content-type"],
+          corsExposedHeaders = Nothing,
+          corsMaxAge = Nothing,
+          corsVaryOrigin = True,
+          corsRequireOrigin = False,
+          corsIgnoreFailures = True
+        }
+
 app :: Env -> Application
-app env = cors (const $ Just CorsResourcePolicy {
-  corsOrigins = Just (["http://127.0.0.1:3000"], True),
-  corsMethods = ["POST"],
-  corsRequestHeaders = ["content-type"],
-  corsExposedHeaders = Nothing,
-  corsMaxAge = Nothing,
-  corsVaryOrigin = True,
-  corsRequireOrigin = False,
-  corsIgnoreFailures = True
-  }) . serve api . hoistServer api (nt env) $ server
+app env = corsMiddleware . serve api . hoistServer api (nt env) $ server
 
 main :: IO ()
 main = do
   createDirectoryIfMissing True . takeDirectory $ databaseFile
 
-  dbPool <- newPool $ defaultPoolConfig (open databaseFile) close 30 10
```

Comment:
The `persistent` backend manages its own resource pool, so no need to use the `resource-pool` api directly anymore (in fact I've removed it as an explicit dependency of the project, since we don't use it directly).